graylog2-server/src/main/java/org/graylog2/migrations/V20170607164210_MigrateReopenedIndicesToAliases.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.migrations;
+
+import com.github.zafarkhaja.semver.Version;
+import com.google.common.collect.ImmutableSet;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.cluster.State;
+import org.graylog2.indexer.ElasticsearchException;
+import org.graylog2.indexer.IndexSet;
+import org.graylog2.indexer.MongoIndexSet;
+import org.graylog2.indexer.cluster.Node;
+import org.graylog2.indexer.cluster.jest.JestUtils;
+import org.graylog2.indexer.indexset.IndexSetService;
+import org.graylog2.indexer.indices.Indices;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.graylog2.indexer.gson.GsonUtils.asJsonObject;
+import static org.graylog2.indexer.gson.GsonUtils.asString;
+
+public class V20170607164210_MigrateReopenedIndicesToAliases extends Migration {
```

Comment:
This class does not log anything, neither failures nor successes of its operation, making it very difficult to debug failures.

Also, while I appreciate the conciseness using streams and maps, they can be very difficult to debug, especially when they transform data which happens to be of an unexpected format, like it appears to be the case below. At the very least these methods should log something about their state when returning default values to aid debugging.